main.tf

```diff
@@ -94,3 +94,114 @@ resource "aws_instance" "this" {
     cpu_credits = local.is_t_instance_type ? var.cpu_credits : null
   }
 }
+
+resource "aws_spot_instance_request" "this" {
+  count = var.instance_request_type == "spot" ? var.instance_count : 0
```

Comment:
do we need a separate resource for spot? can we just provide `null` as a default value for `spot_price`, `wait_for_fulfillment`, `spot_type`, `instance_interruption_behaviour`, `block_duration_minutes`, `valid_from`, and `valid_to`? if you want to use spot you, provide these values, otherwise the default will stay as on-demand?